diag_manager/fms_diag_axis_object.F90

```diff
@@ -384,6 +477,39 @@ subroutine check_if_valid_edges(edges)
        call mpp_error(FATAL, "diag_axit_init: The edge axis has not been defined. "&
                              "Call diag_axis_init for the edge axis first")
   end subroutine check_if_valid_edges
+
+  !> @brief Loop through a variable's axis_id to determine and set the fms2_io filobj type to use
+  !!        and return the domain
+  subroutine determine_the_fileobj_type(axis_id, fileobj_type, domain, var_name)
```

Comment:
Where is this being called?  

Why is `domain` a pointer?  I'm not 100% sure it's thread safe because of the pointer (I think it is thread safe, but I am just not totally positive).  

The description says "to determine and **set** the fms2_io filobj type to use" but it uses `axis_obj(j)%fileobj_type` which is already set.  So nothing is being set here.

This doesn't return a fileobj type; it returns an integer.  If it didn't have the domain pointer as an `intent (out)` argument, this could be a pure integer function instead of a subroutine.

Reply:
> Where is this being called?
This is going to be called in `register_diag_field`, the field has an array of axis ids, this function is going to return the domain and the "fileobj_type" (the type of fms2_io fileobj the variable needs). This then has to be passed on to the diag_files. so we can know what fms2_io obj to open and we have the domain to open the file with. 
I think I should rename "fileobj_type" to "type_of_domain"

> Why is domain a pointer? I'm not 100% sure it's thread safe because of the pointer (I think it is thread safe, but I am just not totally positive).
This is going to be called the init, so there is no openmp(?). If it is not a pointer, then it has to be a copy ...

> The description says "to determine and set the fms2_io filobj type to use" but it uses axis_obj(j)%fileobj_type which is already set. So nothing is being set here.
what if it says "to determine and return the domain and domain_type to use"

Reply:
OK, so what happens to this domain pointer?  Is the domain pointer you pass in part of the diag_object?  I think if you had the actual call to this routine somewhere (even commented out for now), it would be more helpful for me to understand it.